File: water-area/brute-force.js
const greatestVolumePair = (nums) => {
    let largestArea = 0;

    for (let p1 = 0; p1 < nums.length - 1; ++p1) {
        for (let p2 = 0; p2 < nums.length; ++p2) {
            const length = p2 - p1;
            const width = Math.min(nums[ p1 ], nums[ p2 ]);
            largestArea = Math.max(largestArea, length * width);
        }
    }

    return largestArea;
}

console.log(greatestVolumePair([ 7, 1, 3, 9 ]));

File: water-area/brute-force.py
def greatest_volume_pair(nums):
    largest_area = 0

    for p1 in range(len(nums) - 1):
        for p2 in range(len(nums)):
            length = p2 - p1
            width = min(nums[p1], nums[p2])
            largest_area = max(largest_area, length * width)

    return largest_area


print(greatest_volume_pair([7, 1, 3, 9]))
